memory_ttl.go

```diff
@@ -41,23 +44,48 @@ func NewMemoryWithTTL(ttl time.Duration) *MemoryTTL {
 
 // StartGC starts the garbage collection process in a go routine
 func (r *MemoryTTL) StartGC(gcInterval time.Duration) {
-	r.gcInterval = gcInterval
+	if gcInterval <= 0 {
+		return
+	}
+
+	r.gcTicker = time.NewTicker(gcInterval)
+	r.done = make(chan struct{})
+
 	go func() {
-		for _ = range time.Tick(gcInterval) {
-			for key := range r.cache.items {
-				if !r.isValid(key) {
-					r.Delete(key)
+		for {
+			select {
+			case <-r.gcTicker.C:
+				now := time.Now()
+
+				r.Lock()
```

Comment:
starvation is a wrong term - excessive memory usage you used is better - I was thinking about cases when we interrupt sweeping often, but then thought another goroutine that reads all the items concurrently and prepares a new list of expired items can solve that